linker/jvm/src/main/scala/org/scalajs/linker/analyzer/Platform.scala

```diff
@@ -0,0 +1,20 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.analyzer
+
+import scala.collection.mutable
+import scala.collection.concurrent.TrieMap
+
+private[analyzer] object Platform {
+  def emptyThreadSafeMap[K, V]: mutable.Map[K, V] = TrieMap.empty
```

Comment:
IIRC, the TrieMap will not compact the removed entry, so the mememory usage will be lager than the ConcurrentHashMap, but the ConcurrentHashMap have a different trait.

Reply:
We don't remove anything from these maps, so that doesn't seem like a relevant concern.

Reply:
FWIW: TrieMap supports lock-free, "opportunistic" getOrElseUpdate (might create the value multiple times, but won't if it's already in the map). `ConcurrentHashMap` doesn't (`computeIfAbsent`). That was the primary motivator to use `TrieMap`.

Reply:
I always use a `get` to check before invoke the `computeIfAbsnet`, and the bug you mentioned about ConcurrentHashMap has been fixed in JDK 9.

Reply:
I'm not sure what you mean with "the bug I mention".  The JavaDoc is very clear that `computeIfAbsent` can be blocking:

> The entire method invocation is performed atomically, so the function is applied at most once per key. Some attempted update operations on this map by other threads may be blocked while computation is in progress, so the computation should be short and simple, and must not attempt to update any other mappings of this map.

While this can be desirable sometimes, it's not what we want here: We do not need the guarantee that the method invocation is performed atomically, but we do want the operation to remain lock-free.